modules/apps/portal-search/portal-search-rest-impl/src/main/java/com/liferay/portal/search/rest/internal/facet/FacetResponseProcessor.java

```diff
@@ -183,6 +183,10 @@ private String _getFolderDisplayName(
 	private String _getSiteDisplayName(long groupId, Locale locale) {
 		Group group = _groupLocalService.fetchGroup(groupId);
 
+		if (group == null) {
```

Comment:
this change is necessary to make the `testPostSearchPageWithSiteFacetConfiguration` pass. I put it to return blank because I usually prefer to use blank instead of null when it's a string. But it would work with a null value as well.

Reply:
there is a term collector with a value 0 in the term field. It makes the `_groupLocalService.fetchGroup(groupId)` returns null.
see the index 3 in the list in the image
![image](https://github.com/liferay-search/liferay-portal/assets/72256392/a52bab4e-0478-434b-8710-a9e016084be0)

Reply:
@AlmirFe please consider the approach again. If there's corrupt / invalid data, is it good to silently ignore it and pass the problem to the upper layers, like UI? Doing that way, the problem not only creates further problems, but also becomes harder to detect.

If the test is creating a situation that makes no sense to the test itself, then the test needs a fix. 

If this situation should be handled more gracefully in the response processor, which might be a good idea, then the term can just be skipped once and for all. However, proper logging should remain. A separate test for that condition could also be considered.